src/MLJInterface.jl

```diff
@@ -182,142 +135,128 @@ function mlj_to_kwargs(model::MLJModelInterface.Supervised)
 
 end
 
+function mlj_to_kwargs(model::LGBMClassifier, classes)
 
-# X and y must be untyped per MLJ docs (and probably w too, getting nothing by default is zzz though)
+    num_class = length(classes)
+    if model.objective == "binary"
+        if num_class != 2
+            throw(ArgumentError("binary objective number of classes $num_class greater than 2"))
+        end
+        # munge num_class for LightGBM
+        num_class = 1
+    end
+
+    retval = Dict{Symbol, Any}(
+        name => getfield(model, name)
+        for name in fieldnames(typeof(model))
+    )
+
+    retval[:num_class] = num_class
+    return retval
+
+end
+
+# X and y and w must be untyped per MLJ docs
 function fit(mlj_model::MODELS, verbosity::Int, X, y, w=Vector{AbstractFloat}())
 
     # MLJ docs are clear that 0 means silent. but 0 in LightGBM world means "warnings"
     # and < 0 means fatal logs only, so we put intended silence to -1 (which is probably the closest we get)
     verbosity = if verbosity == 0; -1 else verbosity end
 
     y_lgbm, classes = prepare_targets(y, mlj_model)
-    model = model_init(mlj_model)
+    model = model_init(mlj_model, classes)
     X = MLJModelInterface.matrix(X)
     # The FFI wrapper wants Float32 for these
     w = Float32.(w)
-    report = LightGBM.fit(model, X, y_lgbm; verbosity=verbosity, weights=w)
+    report = LightGBM.fit!(model, X, y_lgbm; verbosity=verbosity, weights=w)
 
     model = (model, classes)
     cache = nothing
     report = (report,)
 
     # Caution: The model is a pointer to a memory location including its training data
-    # which is probably bum ... fix in LightGBM.jl?
+    # This definitely needs fixing
```

Comment:
Make a separate issue?

Reply:
https://github.com/IQVIA-ML/LightGBM.jl/issues/1

With a comment from thibaut in there that I seem to have missed..